client/boot/common.js

```diff
@@ -437,10 +433,55 @@ const boot = ( currentUser, registerRoutes ) => {
 	} );
 };
 
-export const bootApp = ( appName, registerRoutes ) => {
+function waitForCookieAuth( user ) {
+	const timeoutMs = 1500;
+	const loggedIn = user.get() !== false;
+	const ipc = require( 'electron' ).ipcRenderer;
+
+	const promiseTimeout = ( ms, promise ) => {
+		const timeout = new Promise( ( _, reject ) => {
+			const id = setTimeout( () => {
+				clearTimeout( id );
+				reject( `Request timed out in ${ ms } ms` );
+			}, ms );
+		} );
+
+		return Promise.race( [ promise, timeout ] );
+	};
+
+	const renderPromise = () => {
+		return new Promise( function ( resolve ) {
```

Comment:
This doesn't need to be wrapped in a function. Just assigning to a variable works, too:
```js
const renderPromise = new Promise( ( resolve ) => {
  // ...
} );
return promiseTimeout( timeoutMs, renderPromise );
```